src/export/Controller/Adminhtml/Exports/Data.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+namespace Aligent\FredhopperExport\Controller\Adminhtml\Exports;
+
+use Aligent\FredhopperExport\Api\Data\ExportInterface;
+use Aligent\FredhopperExport\Model\Data\Export;
+use Aligent\FredhopperExport\Model\Data\ExportFactory;
+use Aligent\FredhopperExport\Model\ResourceModel\Data\Export as ExportResource;
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\App\Response\Http\FileFactory;
+use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\Filesystem\Io\File as Filesystem;
+use Psr\Log\LoggerInterface;
+
+class Data extends Action
+{
+
+    public const string ADMIN_RESOURCE = 'Aligent_FredhopperExport::manage';
+
+    /**
+     * @param Context $context
+     * @param ExportFactory $exportFactory
+     * @param ExportResource $exportResource
+     * @param FileFactory $fileFactory
+     * @param LoggerInterface $logger
+     * @param Filesystem $filesystem
+     */
+    public function __construct(
+        readonly Context $context,
+        private readonly ExportFactory $exportFactory,
+        private readonly ExportResource $exportResource,
+        private readonly FileFactory $fileFactory,
+        private readonly LoggerInterface $logger,
+        private readonly Filesystem $filesystem
+    ) {
+        parent::__construct($context);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(): ResponseInterface
+    {
+        $exportId = (int)$this->getRequest()->getParam('id');
+        if (empty($exportId)) {
+            $this->getMessageManager()->addErrorMessage(__('Export ID is required.'));
+            $this->_redirect('*/*/index');
+            return $this->_response;
+        }
+        $isZip = (bool)$this->getRequest()->getParam('zip');
+        /** @var Export $export */
+        $export = $this->exportFactory->create();
+        $this->exportResource->load($export, $exportId);
+        if ($export->isEmpty()) {
+            $this->getMessageManager()->addErrorMessage(__('Could not load export data.'));
+            $this->_redirect('*/*/index');
+            return $this->_response;
+        }
+        $directory = $export->getDirectory();
+        $filename = match ($export->getExportType()) {
```

Comment:
:open_mouth: `match`. Looks a lot cleaner than trying to do this using IFs or switch cases.